server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -827,8 +829,24 @@ public Relation isFieldWithinQuery(
             QueryRewriteContext context
         ) throws IOException {
             if (isIndexed() == false && pointsMetadataAvailable == false && hasDocValues()) {
-                // we don't have a quick way to run this check on doc values, so fall back to default assuming we are within bounds
-                return Relation.INTERSECTS;
+                if (hasDocValuesSkipper() == false) {
```

Comment:
Did you make this change because a test failed or as an optimization? Asking because I don't expect this change to be required.

Reply:
If not required, I'm doubting whether we should include this particular change now. I prefer to do this change in isolation and add the necessary tests for this.

Reply:
There was a failing test, `org.elasticsearch.index.shard.SearchIdleIT.testSearchIdleBoolQueryMatchOneIndex`

Reply:
I did add tests to `DateFieldTypeTests.java` to test `isFieldWithinQuery` when the doc values skipper is enabled. But it makes sense to me to make this change in isolation. If you want, I can break this out into a separate PR and just mute the test for now

Reply:
> If you want, I can break this out into a separate PR and just mute the test for now

Maybe change `testSearchIdleBoolQueryMatchOneIndex(...)` test to set `index.mapping.use_doc_values_skipper` setting to false if `doc_values_skipper` feature flag is enabled? Then I think the test should pass without adding this logic.

Then we can add this logic with associated tests in a follow up?

Reply:
Ok, follow-up PR #123930 opened.